### PR TITLE
Update repository

### PR DESCRIPTION
**Pull request by rvsia**
- update dependencies
- use webpack-proxy
- remove test.webpack.config
- remove unused scripts
- update to node16

cc @Hyperkid123

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -87,29 +83,22 @@
...
     "extract:messages": "npx @formatjs/cli extract 'src/**/*.{js,jsx}' --out-file ./translations/messages.json",
     "lint": "yarn eslint ./src",
-    "prod": "NODE_ENV=production webpack serve --config config/dev.webpack.config.js",
-    "server:ctr": "node src/server/generateServerKey.js",
+    "prod": "NODE_ENV=production webpack serve --config config/prod.webpack.config.js",
     "start": "NODE_ENV=development webpack serve --config config/dev.webpack.config.js",
     "start:beta": "NODE_ENV=development BETA=true webpack serve --config config/dev.webpack.config.js",
-    "start:proxy": "NODE_ENV=development API_ENDOINT=https://ci.cloud.redhat.com webpack serve --config config/dev.webpack.config.js",
     "test": "jest --passWithNoTests --runInBand",
     "build:prod": "NODE_ENV=production webpack --config config/prod.webpack.config.js",
```
Comment (Hyperkid123) [inline package.json:100]: I believe this one is not used as well. The custom release only calls the `build` script

File: package.json [hunk 2]
```diff
@@ -87,29 +83,22 @@
...
     "redux-mock-store": "^1.5.4",
-    "webpack": "^5.30.0",
-    "webpack-bundle-analyzer": "^4.4.0",
-    "webpack-cli": "^4.6.0",
-    "xhr-mock": "^2.5.1"
+    "webpack-bundle-analyzer": "^4.4.0"
   },
   "scripts": {
     "analyze": "NODE_ENV=production webpack --config config/prod.webpack.config.js --env analyze=true",
     "build": "webpack --config config/dev.webpack.config.js",
```
Comment (Hyperkid123) [inline package.json:93]: Could you add the `NODE_ENV=production` variable to the build script?

File: package.json [hunk 3]
```diff
@@ -2,27 +2,26 @@
   "name": "ansible-dashboard",
   "version": "1.0.0",
   "private": false,
+  "engines": {
+    "node": ">=15.0.0",
```
Comment (Hyperkid123) [inline package.json:6]: Use node 16, please. 15 is no longer supported

File: package.json (hunk 1 above)
Comment (rvsia) [reply package.json]: updated

**Review by Hyperkid123 — APPROVED** [review]

**Pull request merged**